src/async_block_store.rs

```diff
@@ -145,7 +152,64 @@ impl<U: Unblocker> AsyncBlockStore<U> {
         f: impl FnOnce(&mut BlockStore) -> T + Send + 'static,
     ) -> BoxFuture<T> {
         let inner = self.inner.clone();
-        self.unblocker
+        self.runtime
             .unblock(move || f(DerefMut::deref_mut(&mut inner.lock().unwrap())))
     }
+
+    /// A gc loop that runs incremental gc in regular intervals
+    ///
+    /// Gc will run as long as this future is polled. GC is a two step process. First, the
+    /// metadata of expendable non-pinned blocks will be deleted, then the actual data will 
+    /// be removed. This will run the first step and the second step interleaved to minimize
+    /// gc interruptions.
+    pub async fn gc_loop(self, config: GcConfig) -> crate::Result<()> {
+        // initial delay so we don't start gc directly on startup
+        self.runtime.sleep(config.interval / 2).await;
+        loop {
+            debug!("gc_loop running incremental gc");
+            self.incremental_gc(config.min_blocks, config.target_duration)
+                .await?;
+            self.runtime.sleep(config.interval / 2).await;
+            debug!("gc_loop running incremental delete orphaned");
+            self.incremental_delete_orphaned(config.min_blocks, config.target_duration)
+                .await?;
```

Comment:
I take it that we’re either completely sure that this will only error in fatal cases or assume that our caller will restart the loop, right?